### PR TITLE
Auth branch

### PR DESCRIPTION
**Pull request by jxw7410**
Review Auth Branch

### CODE REVIEW

File: db/migrate/20190711164655_add_valid_email_account.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+class AddValidEmailAccount < ActiveRecord::Migration[5.2]
+  def change
+    add_column :users, :validated_account, :boolean, default: false
+  end
+end
```
Comment (yongbingao) [inline db/migrate/20190711164655_add_valid_email_account.rb:5]: adding index on email will probably help with lookup

File: app/controllers/api/users_controller.rb [hunk 2]
```diff
@@ -0,0 +1,19 @@
+class Api::UsersController < ApplicationController
+
+    def create 
+        user = User.new(user_params)
+        if user.save
+            login(user)
+            render json: ['User Logged In'], status: 200
```
Comment (yongbingao) [inline app/controllers/api/users_controller.rb:7]: change variable user to @user

File: app/controllers/api/sessions_controller.rb [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+            login(user)
+            render json: ['User logged in'], status: 200
+        else 
+            render json: ['Log in failed'], status: 422
+        end
+    end
+
+
+    def destroy
+        logout
```
Comment (yongbingao) [inline app/controllers/api/sessions_controller.rb:16]: render some message that user is logged out

**Comment by iRzLi** [comment]
Inside your login you set current_user to user and then reset the session token, won't that make it so your session[:session_token] be different from your current_user.session_token?

**Review by yongbingao — CHANGES_REQUESTED** [review]
@jxw7410 take a look

**Pull request merged**